Lib/idlelib/run.py

```diff
@@ -420,26 +425,22 @@ def isatty(self):
         return True
 
 
-class PseudoOutputFile(PseudoFile):
+class StdOutFile(StdioFile):
 
     def writable(self):
         return True
 
     def write(self, s):
         if self.closed:
             raise ValueError("write to closed file")
-        if type(s) is not str:
-            if not isinstance(s, str):
-                raise TypeError('must be str, not ' + type(s).__name__)
-            # See issue #19481
-            s = str.__str__(s)
+        s = str.encode(s, self.encoding, self.errors).decode(self.encoding)
         return self.shell.write(s, self.tags)
 
 
-class PseudoInputFile(PseudoFile):
+class StdInFile(StdioFile):
 
     def __init__(self, shell, tags, encoding=None):
-        PseudoFile.__init__(self, shell, tags, encoding)
+        StdioFile.__init__(self, shell, tags, encoding)
```

Comment:
```suggestion
        StdioFile.__init__(self, shell, tags, encoding, errors)
```
Pass on the new error parameter.  Ditto comment above about tags.